compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1252,49 +1252,49 @@ TR::Register *OMR::X86::TreeEvaluator::caddEvaluator(TR::Node *node, TR::CodeGen
       }
 
    TR::Register * testRegister = secondChild->getRegister();
-   if (secondChild->getOpCodeValue() == TR::cconst &&
-       testRegister == NULL         &&
-       performTransformation(comp, "O^O CaddEvaluator: checking that the store has not happened yet. Target register:  %x\n", testRegister))
+   if (secondChild->getOpCodeValue() == TR::sconst &&
+      testRegister == NULL         &&
+      performTransformation(comp, "O^O CaddEvaluator: checking that the store has not happened yet. Target register:  %x\n", testRegister))
+   {
+   int32_t value  = secondChild->getConst<uint16_t>();
+   if (!isMemOp)
+      targetRegister = cg->evaluate(firstChild);
+   if (targetRegister && firstChild->getReferenceCount() > 1)
       {
-      int32_t value  = secondChild->getConst<uint16_t>();
-      if (!isMemOp)
-         targetRegister = cg->evaluate(firstChild);
-      if (targetRegister && firstChild->getReferenceCount() > 1)
-         {
-         TR::MemoryReference  *tempMR = generateX86MemoryReference(targetRegister, value, cg);
-         targetRegister = cg->allocateRegister();
-         generateRegMemInstruction(LEA4RegMem, node, targetRegister, tempMR, cg);
-         }
-      else
+      TR::MemoryReference  *tempMR = generateX86MemoryReference(targetRegister, value, cg);
+      targetRegister = cg->allocateRegister();
+      generateRegMemInstruction(LEA4RegMem, node, targetRegister, tempMR, cg);
+      }
+   else
+      {
+      if (value >= 0 && value <= 127)
          {
-         if (value >= 0 && value <= 127)
+         if (value == 1)
```

Comment:
I'm not sure this is right - the sconst is read as unsigned - isn't it possible for the value to be between 128 and 255 inclusive. I know the node is now sconst but that doesn't change the range when the const is still read as a uint16_t...